internal/controller/postgrescluster/postgres.go

```diff
@@ -41,6 +42,35 @@ import (
 	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
 )
 
+// generatePostgresHBAs produces the HBA rules for cluster that incorporates,
+// from highest to lowest precedence:
+//  1. mandatory rules determined by controllers
+//  2. rules in cluster.spec.patroni.dynamicConfiguration
+//  3. default rules, when none were in cluster.spec
+func (*Reconciler) generatePostgresHBAs(
+	ctx context.Context, cluster *v1beta1.PostgresCluster,
+) *postgres.OrderedHBAs {
+	builtin := postgres.NewHBAs()
+	pgmonitor.PostgreSQLHBAs(ctx, cluster, &builtin)
+	pgbouncer.PostgreSQL(cluster, &builtin)
+
+	// Postgres processes HBA rules in order. Start with mandatory rules
+	// so connections are matched against them first.
+	result := new(postgres.OrderedHBAs)
+	result.Append(builtin.Mandatory...)
+
+	// Append any rules specified in the the Patroni section.
+	before := result.Length()
+	result.AppendUnstructured(patroni.PostgresHBAs(cluster.Spec.Patroni)...)
+
+	// When there are no specified rules, include the recommended defaults.
+	if result.Length() == before {
+		result.Append(builtin.Default...)
+	}
```

Comment:
So if the user provides anything, we don't set any defaults? Do we not want to provide defaults for rules they didn't touch?

Reply:
Yep, same as before: https://github.com/CrunchyData/postgres-operator/blob/22f7989f1a1a83ed1dbc64ae8e9b0f6404c54079/internal/patroni/config.go#L261-L266